cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTableAdmin.java

```diff
@@ -186,6 +224,26 @@ private List<String> getCreateIndexStatements(StructuredTableId tableId, Set<Str
     return statements;
   }
 
+  private String getAddColumnStatement(StructuredTableId tableId,
+                                       DatabaseMetaData metaData,
+                                       List<FieldType> nonPrimaryKeyFields)
+    throws SQLException {
+    boolean newColumnsAdded = false;
+    StringJoiner addColumnsStatement = new StringJoiner(", ",
+                                                  "ALTER TABLE " + tableId.getName() + " ",
+                                                  ";");
+    for (FieldType field : nonPrimaryKeyFields) {
+      ResultSet rs = metaData.getColumns(null, null, tableId.getName(), field.getName());
```

Comment:
You shouldn't call this inside the for-loop, as I believe it has to hit the DB on every call. It's better to get the whole table information before this for-loop.

Reply:
updated to call once